hy/macros.py

```diff
@@ -1,3 +1,5 @@
+# fmt: off
```

Comment:
Delete this

Reply:
Are you going to object to every single one of my PRs if I try to fix spacing in it, no matter how unreasonable Black's choices are?

Reply:
Black's changes are not unreasonable. Your code introduced 2 additional [pep8 errors](https://www.python.org/dev/peps/pep-0008/#indentation) that black would have otherwise caught and was the whole point of #2209. If you'd like an approval for this PR you can remove this and I'll happily continue the review as per usual.

Reply:
Please answer the question.

Reply:
I'll object to any PR that attempts to circumvent #2209 by turning off formatting, yours or otherwise.

Reply:
I'm having flashbacks to 2018 when Matthew got fixated on the indentation of my Hy code. Strong words were said and he eventually decided to make his own Lisp-on-a-Python instead (with blackjack and hookers). It wasn't a high point for the project.

Now, I want to emphasize that:

1. I'm not trying to move away from the Black style for a large majority of the Python code. I'd estimate that less than 1% of Hy's Python code, counting by lines, needs to be changed. A lot of Black's changes were pointless but harmless, like changing all the single quotes to double quotes; all I'm trying to revert now is the changes that are actually bad.
2. I'm not asking you to do the work of finding all these mistakes and cleaning them up. I'm only asking you to not forbid me from doing it.
3. No matter what formatting decisions I make, you can always run Black locally with a single command and see the code in the official Black-approved style. I'm not taking it away from you.

I don't think this is a lot to ask, and I don't know why you're being fanatical about this of all issues, especially after over a year of close work with you on Hy during which you've shown good judgment on many occasions.

Can't we broker some kind of compromise? I would happily prioritize work on some difficult Hy feature or bug that's important to you if it would help sway you.

Reply:
@allison-casey Let's wrap this up.

Reply:
It is not a fanatical position. It's a common industry practice with proven benefits. Across Pandas, Scikit-learn, and Django, and their 1,287,351 lines of code, `fmt: off` shows up *zero* times. Black's choices are well founded and it's changes ensure consistent pep8 compliant code in all cases. 

@brandonwillard as the other approver of #2209, what are you feelings on kodi using `fmt: off` to adjust portions of the codebase away from the black standard?

Reply:
Just because a lot of people do something doesn't mean it's beneficial, let alone that any supposed benefits have been proven. Justifying Black's choices in terms of PEP 8 compliance only passes the buck to PEP 8, which would be no less unreasonable to rigidly adhere to.

I'd be happy to show you instances in which Black's choices aren't well founded, but there's no point in me doing that if you've already made up your mind that any deviation from Black's style, however slight, rare, or justified, is unacceptable, no matter how easy it is for you to run Black locally.

Reply:
> @brandonwillard as the other approver of #2209, what are you feelings on kodi using `fmt: off` to adjust portions of the codebase away from the black standard?

Unless something is functionally broken, which it doesn't appear to be in this case, disabling `black` is unwarranted and completely defeats its purpose.

Reply:
If it's not warranted when it makes the code more readable, that seems to imply a belief that consistency with Black's formatting is more important than readability. So then what motivates an interest in consistency? Readability is literally the only thing at stake with respect to formatting, yes?

Reply:
This is one of the snippets you disabled black in order to write:
```diff
# your's
-    source_exports = getattr(source_module, "_hy_export_macros",
-        [k for k in source_macros.keys() if not k.startswith('_')])
# black's
+    source_exports = getattr(
+        source_module,
+        "_hy_export_macros",
+        [k for k in source_macros.keys() if not k.startswith("_")],
+    )
```

pep8 says it's forbidden:
> ![image](https://user-images.githubusercontent.com/18231154/155250569-7c6db237-516c-4d58-a0db-b44a35cfd41f.png)

Google says it's forbidden:
> ![image](https://user-images.githubusercontent.com/18231154/155250628-4105db5f-8010-41f0-b6b8-c3c8cafe9c6f.png)

Google dislikes it so much they wrote about it as the **Rectangle Rule**
> ![image](https://user-images.githubusercontent.com/18231154/155255000-4aa07566-0d9b-4ccf-a941-5a602114470e.png)

Which underpins their wrapping rules for Python, Java, and Javascript and you'll see enforced in their repos with `yapf/black`, `google-java-format`, and facebook's `prettier` respectively

Clojure says arguments should be vertically aligned and CIDER and Emac's indent region will fight me if i attempt to push an arg before it's indent block
> ![image](https://user-images.githubusercontent.com/18231154/155252092-26cad64a-2a38-42c3-aa17-1522d755da89.png)

Hy *itself* says it's "BAD. And evil."
> ![image](https://user-images.githubusercontent.com/18231154/155252246-8366172c-4d2e-4020-8adc-f6a25c8cb10f.png)

Black's choices follow language agnostic standard practice for writing readable code. They may not be your preferred style, but they're certainly not "bad" or "unreasonable". Brandon and I are in agreement then that nothing is obviously broken, and so the use of `fmt: off` is unwarranted. If you'd like to continue with this PR, I would be happy to do a full review once this line has been removed.

Reply:
I think Black's choice in that case is fine, actually. It's not what I wrote, but it's perfectly reasonable.

I don't think style guides are to your point so long as your position is that we have to do what Black says no matter what. The only "style guide" whose authority you recognize in that case is the one implicitly defined by Black. Otherwise, it seems we're in agreement that we don't care for style guides.

Hy's style guide is not Hy itself (thank goodness). It's something I've had on my mental chopping block for a long time, although I've yet to do the big documentation overhaul that would involve its removal.

> If you'd like to continue with this PR, I would be happy to do a full review once this line has been removed.

But won't you then just oppose the PR because it doesn't pass the Black check, as well as all future PRs with any spacing fixes, no matter what? So then it looks like your bigger issue still needs to be resolved. You're more than welcome to review this PR in the meantime, of course.

Reply:
format the file with black, it will pass CI, and I will approve the PR assuming there's nothing else that needs addressing. you can format the file before each commit or at the very end, CI does not care.

Reply:
But you're still going to object to every single PR with spacing fixes no matter what. I want to negotiate, but in order for that to happen, you need to be willing to negotiate.

Reply:
black handles everything. there are no spaces to fix.